src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncSpillTests.cs

```diff
@@ -3888,5 +3887,2228 @@ class AltBoolean
+    IL_0061:  stloc.2
+    IL_0062:  ldarg.0
+    IL_0063:  ldflda     ""System.Runtime.CompilerServices.TaskAwaiter<int> Program.<Assign>d__0.<>u__1""
+    IL_0068:  initobj    ""System.Runtime.CompilerServices.TaskAwaiter<int>""
+    IL_006e:  ldarg.0
+    IL_006f:  ldc.i4.m1
+    IL_0070:  dup
+    IL_0071:  stloc.0
+    IL_0072:  stfld      ""int Program.<Assign>d__0.<>1__state""
+    IL_0077:  ldloca.s   V_2
+    IL_0079:  call       ""int System.Runtime.CompilerServices.TaskAwaiter<int>.GetResult()""
+    IL_007e:  stloc.1
+    IL_007f:  ldarg.0
+    IL_0080:  ldfld      ""A Program.<Assign>d__0.<>7__wrap1""
+    IL_0085:  ldflda     ""B A.b""
+    IL_008a:  ldflda     ""C B.c""
+    IL_008f:  ldloc.1
+    IL_0090:  stfld      ""int C.x""
+    IL_0095:  ldarg.0
+    IL_0096:  ldnull
+    IL_0097:  stfld      ""A Program.<Assign>d__0.<>7__wrap1""
+    IL_009c:  leave.s    IL_00b5
+  }
+  catch System.Exception
+  {
+    IL_009e:  stloc.3
+    IL_009f:  ldarg.0
+    IL_00a0:  ldc.i4.s   -2
+    IL_00a2:  stfld      ""int Program.<Assign>d__0.<>1__state""
+    IL_00a7:  ldarg.0
+    IL_00a8:  ldflda     ""System.Runtime.CompilerServices.AsyncTaskMethodBuilder Program.<Assign>d__0.<>t__builder""
+    IL_00ad:  ldloc.3
+    IL_00ae:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetException(System.Exception)""
+    IL_00b3:  leave.s    IL_00c8
+  }
+  IL_00b5:  ldarg.0
+  IL_00b6:  ldc.i4.s   -2
+  IL_00b8:  stfld      ""int Program.<Assign>d__0.<>1__state""
+  IL_00bd:  ldarg.0
+  IL_00be:  ldflda     ""System.Runtime.CompilerServices.AsyncTaskMethodBuilder Program.<Assign>d__0.<>t__builder""
+  IL_00c3:  call       ""void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetResult()""
+  IL_00c8:  ret
+}");
+        }
+
+        [Fact]
+        [WorkItem(42755, "https://github.com/dotnet/roslyn/issues/42755")]
+        public void KeepLtrSemantics_ClassPropertyAssignmentOnClassProperty()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+
+class Program
+{
+    static async Task Assign(A a)
+    {
+        a.b.x = await Write(""RHS"");
+    }
+
+    static async Task Main(string[] args)
+    {
+        await TestAIsNull();
+        await TestAIsNotNull();
+        await ReassignADuringAssignment();
+    }
+
+    static async Task TestAIsNull()
+    {
+        Console.WriteLine(nameof(TestAIsNull));
+        
+        A a = null;
+        Console.WriteLine(""Before Assignment"");
+        try
+        {
+            await Assign(a);
+        }
+        catch (NullReferenceException)
+        {
+            Console.WriteLine(""Caught NullReferenceException"");
+        }
+    }
+
+    static async Task TestAIsNotNull()
+    {
+        Console.WriteLine(nameof(TestAIsNotNull));
+
+        var a = new A{ b = new B() };
+        Console.WriteLine(""Before Assignment a.b.x is: "" + a.b.x);
```

Comment:
>a.b.x [](start = 61, length = 5)

Please do this check by accessing fields directly, so that we could see when assignments accesses the properties.